internal/pkg/plan/persist.go

```diff
@@ -23,22 +24,19 @@ type persistPlanBuilder struct {
 }
 
 func (b *persistPlanBuilder) build() {
-	// New configs
+	// Process children of the existing objects
 	for _, path := range b.UntrackedDirs() {
-		for _, branch := range b.Branches() {
-			if actions := b.tryAddConfig(path, branch); actions != nil {
-				b.actions = append(b.actions, actions...)
-				break
-			}
+		for _, parent := range b.All() {
+			b.tryAdd(path, parent)
 		}
 	}
 
-	// New config rows from existing configs
+	// Process children of the new objects
 	for _, path := range b.UntrackedDirs() {
-		for _, config := range b.Configs() {
-			if action := b.tryAddConfigRow(path, config.Path(), config.ConfigKey); action != nil {
-				b.actions = append(b.actions, action)
-				break
+		actions := b.actions
```

Comment:
Dalsi `for` je pre pod-objekty novych objektov :) ... tj. v predchadzajucom `for` sa najde napr. novy config, 
... tak tu sa potom hlada, ci ten novy config ma aj (nove) `rows`.